pkg/util/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,55 @@
+package dockerfile
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/builder/parser"
+)
+
+// ParseTreeToDockerfile takes a Dockerfile AST node, as generated by
+// parser.Parse, and returns a new, equivalent, Dockerfile.
+func ParseTreeToDockerfile(node *parser.Node) []byte {
```

Comment:
Kudos for this, I like it. Can you refactor the rest of the builder package to use this? Follow-up is fine with me.